cnf-certification-test/platform/suite.go

```diff
@@ -272,9 +272,23 @@ func testTainted(check *checksdb.Check, env *provider.TestEnvironment) {
 	// Loop through the debug pods that are tied to each node.
 	for _, dp := range env.DebugPods {
 		nodeName := dp.Spec.NodeName
-
 		check.LogInfo("Testing node %q", nodeName)
 
+		// Ensure we are only testing nodes that have CNF workload deployed on them.
+		nodeHasWorkload := false
```

Comment:
Proposal: Move this code to provider->nodes.go. Instead of looping through the debug pods here, loop through the nodes and check if workload is deployed with new function. Then get the debug pod with the env.DebugPods[cut.NodeName] map